picocli-codegen/src/main/java/picocli/codegen/docgen/manpage/ManPageGenerator.java

```diff
@@ -381,6 +381,14 @@ static void genOptions(PrintWriter pw, CommandSpec spec) {
         IParameterRenderer parameterRenderer = spec.commandLine().getHelp().createDefaultParameterRenderer();
 
         List<OptionSpec> options = new ArrayList<OptionSpec>(spec.options()); // options are stored in order of declaration
+
+        // remove hidden options
+        for (Iterator<OptionSpec> iter = options.iterator(); iter.hasNext();) {
+            if (iter.next().hidden()) {
+                iter.remove();
+            }
+        }
```

Comment:
I believe this is the right approach, but we should do this at the top of this method, before rendering the `== Options` header.

Reply:
Ahh, I missed that one, i just pushed an update